modules/resonator.h

```diff
@@ -0,0 +1,184 @@
+#pragma once
+#ifndef DSY_RESONATOR_H
+#define DSY_RESONATOR_H
+
+#include <stdint.h>
+#include <stddef.h>
+#include "dsp.h"
+#ifdef __cplusplus
+
+#define PI_POW_3 PI_F* PI_F* PI_F
+#define PI_POW_5 PI_POW_3* PI_F* PI_F
+
+#define kMaxNumModes 24
+#define kModeBatchSize 4
+
+/** @file resonator.h */
+
+namespace daisysp
+{
+enum FilterMode
+{
+    FILTER_MODE_LOW_PASS,
+    FILTER_MODE_BAND_PASS,
+    FILTER_MODE_BAND_PASS_NORMALIZED,
+    FILTER_MODE_HIGH_PASS
+};
```

Comment:
Move this into ResonatorSvf ("FilterMode" seems like something that may already be in use or will be used in/by other modules later, best to keep it scoped).

Once it's there you'll need to reference it as `Resonator<kModeBatchSize>::FILTER_MODE_LOW_PASS` so you can shorten it down to `LOW_PASS`, etc.

Then you may be able to reference it like this from the resonator:
```
batch_processor->Process<batch_processor->BAND_PASS, true>(
                mode_f, mode_q, mode_a, in, &out);
```

and just as `LOW_PASS`, etc. inside the `ResonatorSvf` class. So the syntax doesn't get too gross inside which is nice.